Lib/test/test_traceback.py

```diff
@@ -3802,6 +3802,26 @@ def test_traceback_header(self):
         exc = traceback.TracebackException(Exception, Exception("haven"), None)
         self.assertEqual(list(exc.format()), ["Exception: haven\n"])
 
+    def test_exception_punctuation_handling_with_suggestions(self):
+        def raise_mssage(message, name):
+            try:
+                raise NameError(message, name=name)
+            except Exception as e:
+                return traceback.TracebackException.from_exception(e)._str
```

Comment:
Can't you use the `list(exc.format())` one instead? it's better than accessing a private attribute.